### PR TITLE
Remove argument separator from commands 

### PR DESCRIPTION
**Pull request by blackdwarf**
Currently, the argument separator (`--`) is only supported on `dotnet run`. This PR removes it from other commands to make the help and usage of these command clearer. 

It also adds a new test to check the `dotnet run` message output and changes an existing test to not check the msbuild argument mesasge for `dotnet run` but only for others. 

/cc @livarcocc @piotrpMSFT @jonsequitur

### CODE REVIEW

**Comment by blackdwarf** [comment]
@piotrpMSFT added you as a reviewer as well.

File: test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs [hunk 1]
```diff
@@ -78,6 +77,21 @@ public void When_help_is_invoked_Then_MSBuild_extra_options_text_is_included_in_
...
+
+            var projectDirectory = TestAssetsManager.CreateTestDirectory("RunContainsAppArgumentsText");
+            var result = new TestCommand("dotnet")
+                .WithWorkingDirectory(projectDirectory.Path)
+                .ExecuteWithCapturedOutput("run --help");
+            
+            result.ExitCode.Should().Be(0);
+            result.StdOut.Should().Contain(AppArgumentsText);
+        }
+
```
Comment (krwq) [inline test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs]: nit: double enter

File: src/dotnet/commands/dotnet-pack/PackCommand.cs [hunk 2]
```diff
@@ -19,8 +19,8 @@ public static int Run(string[] args)
                 Name = "pack",
                 FullName = LocalizableStrings.AppFullName,
                 Description = LocalizableStrings.AppDescription,
-                AllowArgumentSeparator = true,
-                ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-pack/PackCommand.cs]: NIT: get rid of the extra spacing.

File: src/dotnet/commands/dotnet-publish/Program.cs [hunk 3]
```diff
@@ -17,8 +17,8 @@ public static int Run(string[] args)
             app.Name = "dotnet publish";
             app.FullName = LocalizableStrings.AppFullName;
             app.Description = LocalizableStrings.AppDescription;
-            app.AllowArgumentSeparator = true;
-            app.ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText;
+            app.HandleRemainingArguments = true;
+            app.ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText;
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-publish/Program.cs]: NIT: get rid of the extra spacing.

File: src/dotnet/commands/dotnet-restore/Program.cs [hunk 4]
```diff
@@ -19,8 +19,8 @@ public static int Run(string[] args)
                 Name = "restore",
                 FullName = LocalizableStrings.AppFullName,
                 Description = LocalizableStrings.AppDescription,
-                AllowArgumentSeparator = true,
-                ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-restore/Program.cs]: NIT: get rid of the extra spacing.

**Comment by livarcocc** [comment]
@srivatsn

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX x64 Release Build

**Pull request merged**